### PR TITLE
fix: changed calling registered logic to use SIP registration xStatus

### LINKED ISSUE #22: VCQ Macro is failing to correctly identify Webex Calling registered devices

### PR DESCRIPTION
**Pull request by wmpmills**
Move away from checking the Display Name config to using the SIP Registration xStatus for 'Registered' before attempting to place SIP calls.

### CODE REVIEW

File: src/lib/static/sip-macro-template.txt [hunk 1]
```diff
@@ -66,9 +66,9 @@ main()
 
 async function checkCallingRegistered(){
   try{
-    let displayName = await xapi.Config.SIP.DisplayName.get();
-    console.log(displayName);
-    if(displayName.trim().startsWith("+") && displayName.trim().indexOf("internal:") >= 0){
+    let status = await xapi.Status.SIP.Registration[1].Status.get();
```
Comment (askmrsinh) [inline src/lib/static/sip-macro-template.txt]: Is this guaranteed to return a value without throwing an error?

**Comment by wmpmills** [comment]
@askmrsinh , the existing **checkCallingRegistered** function didn't return a value that was used elsewhere in the macro.  I have updated the **checkCallingRegistered** function to always return a value and additionally added logic to the **processWebViews** function to prevent calls when Webex Calling services are not registered. That situation would prevent the macro from getting a usable Remote Number for detecting Call Queue and Agent Answer states.

**Comment by askmrsinh** [comment]
I appreciate the explanation @wmpmills. But my question was more in regards to `await xapi.Status.SIP.Registration[1].Status.get()`. If it's guaranteed to return a value, good; otherwise there should be some exception handling. From your explanation, it looks like it's guaranteed to return some value.

**Comment by wmpmills** [comment]
@askmrsinh , would you mind approving the PR if there are no other changes required. Thanks -Will

**Pull request merged**